src/refiners/foundationals/latent_diffusion/schedulers/euler.py

```diff
@@ -72,13 +72,15 @@ def __call__(
 
         gamma = min(s_churn / (len(self.sigmas) - 1), 2**0.5 - 1) if s_tmin <= sigma <= s_tmax else 0
 
-        alt_noise = torch.randn(noise.shape, generator=generator, device=noise.device, dtype=noise.dtype)
+        alt_noise = torch.randn(
```

Comment:
Now that `noise` is (properly) renamed into `predicted_noise` I would be tempted to use `noise` for this one (stochastic sampler), i.e. the `alt_` prefix is not needed anymore. WDYT?

Reply:
cc @brycedrennan

Reply:
Sounds good to me. 

If I'm reading the code correctly, with default settings, the noise is never used anyways.

Maybe never used at all. https://github.com/huggingface/diffusers/issues/4785#issuecomment-1694233294

Reply:
I've surely gone mad.  Does the whole thing simplify down to:

```python
    def __call__(
        self,
        x: Tensor,
        predicted_noise: Tensor,
        step: int,
        generator: Generator | None = None,
    ) -> Tensor:
        assert self.first_inference_step <= step < self.num_inference_steps, "invalid step {step}"

        delta_t = self.sigmas[step + 1] - self.sigmas[step]
        return x + predicted_noise * delta_t
 ```

Reply:
Disclaimer: I have basically not looked into the Euler scheduler yet (it was a contribution). But given the name it makes sense that it simplifies to this (Euler's method) when you remove the stochastic part (churn = 0).

There is [an in depth explanation about this scheduler](https://isamu-website.medium.com/understanding-k-diffusion-from-their-research-paper-and-source-code-55ae4aa802f) on @isamu-isozaki's blog by the way!

Reply:
Yes very in depth!

Reply:
@catwell Thanks for the exposure!